include/nuttx/spinlock.h

```diff
@@ -49,6 +49,21 @@ typedef uint8_t spinlock_t;
  * SP_LOCKED and SP_UNLOCKED must be constants of type spinlock_t.
  */
 
+#if defined(CONFIG_TICKET_SPINLOCK)
+#include <stdatomic.h>
+
+struct ticket_spinlock_s
```

Comment:
```suggestion
union spinlock_u
{
  struct
  {
    uint16_t owner;
    uint16_t next;
  };
  uint32_t value;
};

typedef struct spinlock_u spinlock_t;
```

Reply:
Why is uint32_t ?
It should be spinlock_t because if we uses aarch64 the spin clock size is 64bit instead of 32bit.

Reply:
Do you want to only use spinlock_t and remove ticket_spinlock?

In 64bit machine spinlock is 64bit, I had tested modify spinlock to 32bit and it can't boot but I have not find reason.
I think this modification will cause same problem.

Reply:
> Do you want to only use spinlock_t and remove ticket_spinlock?
> 

Yes, it's an implementation detail where to use the ticket algorithm, and then shouldn't expose the difference to the caller.  Please consider how Linux switch the spinlock algo, do they add something like ticket_spinlock_s?

> In 64bit machine spinlock is 64bit, I had tested modify spinlock to 32bit and it can't boot because the underlayer of up_testset is 64bit. So I just use first 32bit as ticket spinlock in 64bit machine.
> 

64bit support 32bit atomic too, you can simplify modify the implementation of up_testset on 64bit.

> I think this modification will cause same problem.
> 
> And spinlock_t is already defined at arch/spinlock.h

That's why I suggest you don't include arch/spinlock.h in case of CONFIG_TICKET_SPINLOCK.